.circleci/config.yml

```diff
@@ -1113,6 +1117,54 @@ jobs:
       # running a broken package fails
       - run: "! ${KURTOSIS_BINPATH} run github.com/kurtosis-tech/datastore-army-package@25ecb4c271f870d6fdf8745169a142519e49aab8"
 
+  test_taints_and_tolerations:
+    executor: ubuntu_vm
+    parameters:
+      <<: *param_cli_cluster_backend
+    steps:
+      # need this for something that k3s uses
+      - checkout
+
+      - <<: *abort_job_if_only_docs_changes
+
+      # Set up Kurtosis
+      - attach_workspace:
+          at: "<< pipeline.parameters.workspace-with-cli-binary-and-images-mountpoint >>"
+
+      - run: |
+          echo 'export KURTOSIS_BINPATH="<< pipeline.parameters.workspace-with-cli-binary-and-images-mountpoint >>/<< pipeline.parameters.cli-dist-home-relative-dirpath >>/<< pipeline.parameters.cli-linux-amd-64-binary-relative-filepath >>"' >> "${BASH_ENV}"
+      - run: "${KURTOSIS_BINPATH} analytics disable"
+
+      # When backend is 'kubernetes' install kubernetes and start a Kurtosis gateway
+      - when:
+          condition:
+            equal: [ "kubernetes", << parameters.cli-cluster-backend >> ]
+          <<: *steps_prepare_testing_k8s_k3s
+
+      - when:
+          condition:
+            equal: [ "docker", << parameters.cli-cluster-backend >> ]
+          <<: *steps_prepare_testing_docker
+
+      - when:
+          condition:
+            equal: [ "kubernetes", << parameters.cli-cluster-backend >> ]
+          steps:
+            - run: "sudo snap install kubectl --classic"
+            - run: "kubectl taint nodes k3d-k3s-default-server-0 foo=bar:NoSchedule"
+            # test that without passing toleration the container doesn't get started
+            - run: "! ${KURTOSIS_BINPATH} run --enclave without-tolerations << pipeline.parameters.workspace-with-cli-binary-and-images-mountpoint >>/<< pipeline.parameters.starlark-test-taints-and-tolerations >> --args {\"with_tolerations\": false}"
```

Comment:
I wonder if you could reinforce this test by checking with docker ps (and some labels)? to check that the container was not created, because the kurtosis run could fail for something else and this test will pass ok.

Reply:
If it not takes much time, if it takes much time is ok with current version for me